### ISSUE TITLE
Lag Issue when click and hovering

### ISSUE THREAD

**Issue by LemuelDayrit** [AstraExt/astra-monitor#138]
this is the video proof that has lag issue also im using gnome wayland

https://github.com/user-attachments/assets/af883de0-d997-4797-ad22-a48e6913a196

**Comment by ljuzig** [comment]
Have you installed all dependencies? Please screen the "Welcome" setting page.
Have you set all Data Sources to Auto?

**Comment by LemuelDayrit** [comment]
i have install all dependencies except for amdgpu_top because i use nvidia

![Screenshot from 2024-07-31 15-03-05](https://github.com/user-attachments/assets/3bcadc91-a63d-410a-bae0-884671e97bdb)

**Comment by LemuelDayrit** [comment]
i have install all dependencies except for amdgpu_top because i use nvidia

![Screenshot from 2024-07-31 15-03-05](https://github.com/user-attachments/assets/69b682e8-c029-4b0b-96e7-763421ef40f4)

**Comment by ljuzig** [comment]
Are you still experiencing this issue in the latest `28 (EGOv45)` version?

**Comment by xyven1** [comment]
I'm seeing this issue as well, no GPU, `29 (EGOv46)`, and all sources are present. Overall it just feels like all the updates block the entire gnome-shell instead of processing the updates asynchronously.

**Comment by ljuzig** [comment]
The updates are processed `asynchronously`. Do you have all dependencies met? Do you have all sources on `auto`? Do you have `libgtop` installed?

The fact that opening the menus and move the mouse through them fastly might slightly freeze the interface is a known issue, although I've tried to minimize it as much as possible. However, the extension should not cause any slowdown during data updates if the default configuration is used and all dependencies are installed. This has been thoroughly tested.

**Comment by xyven1** [comment]
@ljuzig yes to all your questions. Absolutely every dependency is installed, and the splash screen says that libgtop is successfully detected. The lag is quite pronounced and effects the entire gnome shell. I'm not sure what's the issue is. I should specify that the lag is only noticable when the menus are open, not when it's just the tooltip.

It may be a deeper issue with the gnome shell, but I havent noticed it until I started using astra.

**Comment by ljuzig** [comment]
@xyven1 what's your gpu? are you on x11 or wayland? While I was testing nvidia monitoring I faced this issue too. I do not have it on my AMD dGPU. I will dig deep into this in the next weeks.

**Comment by xyven1** [comment]
@ljuzig 
Intel iGPU, Wayland. I can send a full system dump later tn.

**Comment by xyven1** [comment]
@ljuzig Its not perfect but seems much much better. Thanks for the fix.

**Comment by xyven1** [comment]
Hmm, after some further testing it appears as though the lag is still there.
